docs/guides/using-with-flake-parts.md

```diff
@@ -0,0 +1,164 @@
+`devenv` can be used without the `devenv` CLI by integrating into [Nix Flakes](https://www.tweag.io/blog/2020-05-25-flakes/), if you're more familiar with the Nix language/ecosystem.
+
+Some usecases for using devenv configuration inside flakes is for projects that want to define other Nix flake features, apart from the development shell.
+These include a Nix package for the project, NixOS and home-manager modules related to the project.
+Usually you want to use the same lock file for the development shell as well as the Nix package and others, so that everything is based on the same nixpkgs.
+
+A Nix flake includes the inputs from `devenv.yaml` as well as the devenv configuration that you'd usually find in `devenv.nix`. `flake.lock` is the lock file for Nix flakes, the equivalent to `devenv.lock`.
```

Comment:
When using `devenv` via the flake module, assuming that the `devenv` CLI isn't available, how would I run processes/services defined in configuration?

Reply:
On that note, it would be pretty amazing to expose the process runners and/or the direct commands they execute via `flake.apps`

Reply:
> assuming that the `devenv` CLI isn't available

Bad assumption -- I just checked out this branch (after rebasing onto `main`) and I see that the CLI is available. However, when I run `devenv up`:

```console
✦ ❯ devenv up
File devenv.nix does not exist. To get started, run:

  $ devenv init
```

Reply:
I accidentally shadowed the shim by adding the whole devenv package to the packages. That was old code from before the shim existed. Now the module is on par with the existing flake integration.